src/hooks/useBalanceChartData/useBalanceChartData.ts

```diff
@@ -311,6 +327,33 @@ export const useBalanceChartData: UseBalanceChartData = args => {
   const balances = useAppSelector(state =>
     selectPortfolioCryptoBalancesByAccountIdAboveThreshold(state, accountId)
   )
+
+  // Get total delegation
+  // TODO(ryankk): consolidate accountSpecifiers creation to be the same everywhere
+  const cosmosCaip2: CAIP2 = caip2.toCAIP2({
+    chain: ChainTypes.Cosmos,
+    network: NetworkTypes.COSMOSHUB_MAINNET
+  })
+
+  const accountSpecifiers = useSelector(selectAccountSpecifiers)
+  const account = accountSpecifiers.reduce((acc, accountSpecifier) => {
```

Comment:
https://github.com/shapeshift/web/blob/develop/src/context/TransactionsProvider/TransactionsProvider.tsx#L39

this logic is now duplicated in at least 3 places i know of, can we please move it to the portfolioSlice utils

Reply:
The logic you linked in TransactionProvider looks different than what we have

Reply:
it's the same idea - my requested change still stands.

Reply:
All the places where I find this logic, its slightly different enough to keep them separate at this time. However, this brings up a bigger problem where `accountSpecifiers` can mean a `CAIP10` or `pubKey` depending on where it's being defined. Here is an example of what I'm talking about.
https://github.com/shapeshift/web/blob/chart_cosmos_fixes2/src/state/slices/accountSpecifiersSlice/selectors.ts#L17
I feel like this is something that can (and needs) to be cleaned up, but possibly could be done in a follow up PR. Thoughts? @0xdef1cafe